### DISCUSSION TITLE
Installation with PHP 8.0.3

### DISCUSSION THREAD

**Discussion by supasteri** [atrocore/atrocore#187] [General]
Hi there

I am following the installation instructions on ubuntu 20.04.1 LTS, with PHP 8.0.3. Unfortunately it would appear that one of the embedded composer dependencies is not detecting this version of PHP as a valid dependency: 

`
atropim@pim:~/tmp$ php composer.phar update
26.07.2021 14:26:12 | Loading composer repositories with package information
26.07.2021 14:26:12 | Updating dependencies
26.07.2021 14:26:15 | Your requirements could not be resolved to an installable set of packages.
26.07.2021 14:26:15 | 
  Problem 1
    - atrocore/core 1.2.9 requires php ^7.1 -> your PHP version (8.0.3) does not satisfy that requirement.
    - atrocore/core 1.2.8 requires php ^7.1 -> your PHP version (8.0.3) does not satisfy that requirement.
    - atrocore/core 1.2.7 requires php ^7.1 -> your PHP version (8.0.3) does not satisfy that requirement.
    - atrocore/core 1.2.6 requires php ^7.1 -> your PHP version (8.0.3) does not satisfy that requirement.
    - atrocore/core 1.2.57 requires php ^7.4 -> your PHP version (8.0.3) does not satisfy that requirement.
    - atrocore/core 1.2.54 requires php ^7.1 -> your PHP version (8.0.3) does not satisfy that requirement.
`

![image](https://user-images.githubusercontent.com/13227259/127005605-49f42102-a1f3-4753-8667-35af92f3511f.png)

Based on the research that I have done, one of the composer JSON files needs to be updated to indicate that 8.0.3 is supported?

Please advise?

**Comment by supasteri** [comment]
Hi guys

Looks like I managed to workaround this issue by using the --ignore-platform-reqs switch on composer.phar (version 1.10)

Rgds
Soteri

**Reply by marcus-at-localhost** [reply]
I think you will run into problems with the atro pim auto update, since you can't set the cli arguments there.

A better way would be to use the `config.platform` option[1] in the top composer.json file 

```json
{
  "require": {
    // requirements
  },
  "config": {
    "platform":{
      "php": "7.4.21",
      "ext-imagick": "3.4.4"
    }
  }
}
```

Omit the `ext-imagick` entry if you have the extension installed. It was just my way to get Atro PIM running on a server that doesn't have ext-imagick installed.[2]

[1] https://getcomposer.org/doc/06-config.md#platform
[2] https://marcus-obst.de/blog/faking-platform-requirements-in-composer

**Comment by rratsun** [comment]
Hello,

We have changed the required PHP version in atrocore 1.2.58 to 7.4.
Now it should work fine.

We recommend to use PHP7.4. Upper version of PHP can work unstable.
